### PR TITLE
API changes

### PR DESCRIPTION
**Pull request by Njs41**
A minor optimization is added, as well as moves to using the IPlayerProvider interface rather than the IServer interface.

### CODE REVIEW

File: src/no/runsafe/nchannels/database/ChannelMembershipRepository.java [hunk 1]
```diff
@@ -25,7 +31,7 @@ public ISchemaUpdate getSchemaUpdateQueries()
 	{
 		ISchemaUpdate update = new SchemaUpdate();
 		update.addQueries(
-			"CREATE TABLE nchannel_members (" +
+			"CREATE TABLE IF NOT EXISTS nchannel_members (" +
```
Comment (mortenn) [inline src/no/runsafe/nchannels/database/ChannelMembershipRepository.java]: Should not be doing this, rather the version is corrected in the database.

File: src/no/runsafe/nchannels/database/ChannelMembershipRepository.java [hunk 2]
```diff
@@ -44,17 +50,23 @@ public ISchemaUpdate getSchemaUpdateQueries()
...
 		return database.queryStrings("SELECT channel FROM nchannel_members WHERE player=?", player.getUniqueId().toString());
 	}
 
-	public List<UUID> getChannelPlayers(String channel)
+	public List<IPlayer> getChannelPlayers(String channel)
 	{
 		List<String> stringIds = database.queryStrings("SELECT player FROM nchannel_members WHERE channel=?", channel);
-		List<UUID> playerIds = new ArrayList<UUID>();
+		List<IPlayer> playerIds = new ArrayList<>();
 		for(String playerStringId : stringIds)
```
Comment (mortenn) [inline src/no/runsafe/nchannels/database/ChannelMembershipRepository.java]: Add a method to IPlayerProvider that converts a List of UUID into a List of IPlayer

File: src/no/runsafe/nchannels/database/ChannelMembershipRepository.java [hunk 3]
```diff
@@ -44,19 +48,20 @@ public ISchemaUpdate getSchemaUpdateQueries()
...
-	public List<UUID> getChannelPlayers(String channel)
+	public List<IPlayer> getChannelPlayers(String channel)
 	{
-		List<String> stringIds = database.queryStrings("SELECT player FROM nchannel_members WHERE channel=?", channel);
-		List<UUID> playerIds = new ArrayList<UUID>();
-		for(String playerStringId : stringIds)
-			playerIds.add(UUID.fromString(playerStringId));
-
-		return playerIds;
+		return playerProvider.getPlayersByIDs(database.queryStrings("SELECT player FROM nchannel_members WHERE channel=?", channel));
```
Comment (mortenn) [inline src/no/runsafe/nchannels/database/ChannelMembershipRepository.java]: database.queryPlayers()? ^_^

Comment (Njs41) [reply src/no/runsafe/nchannels/database/ChannelMembershipRepository.java]: I think that gets players using their usernames, so that might not work here

Comment (mortenn) [reply src/no/runsafe/nchannels/database/ChannelMembershipRepository.java]: I meant to add a new method.. but you could change that one to be smart about what it treats values as.. it's going to be either username or uuid and it should all become uuid over time anyway.

Comment (Njs41) [reply src/no/runsafe/nchannels/database/ChannelMembershipRepository.java]: Can do

**Pull request merged**